lib/fastlane/actions/clear_derived_data.rb

```diff
@@ -1,8 +1,14 @@
 module Fastlane
   module Actions
     class ClearDerivedDataAction < Action
-      def self.run(params)
-        path = File.expand_path("~/Library/Developer/Xcode/DerivedData")
+      def self.run(options)
+        if options[:derived_data_path]
```

Comment:
You can revert these changes once you provide a `default_value` in the `options` array. Make sure to still run `File.expand_path(options[: derived_data_path])` when accessing the actual value.